tools/waveica/waveica_wrapper.R

```diff
@@ -1,4 +1,57 @@
-waveica <- function(data,
+read_csv <- function(file, metadata) {
+  if (!is.na(metadata)) {
+    ft_table <- read.csv(file, header = TRUE)
+    mt_data <- read.csv(metadata, header = TRUE)
+    data <- merge(mt_data, ft_table, by = "sampleName")
+  } else {
+    data <- read.csv(file, header = TRUE)
+  }
+
+  return(data)
+}
```

Comment:
Do we need `else` here? It seems to me that this can be done a bit cleaner:
```suggestion
read_csv <- function(file, metadata) {
  data <- read.csv(file, header = TRUE)
  if (!is.na(metadata)) {
    mt_data <- read.csv(metadata, header = TRUE)
    data <- merge(mt_data, data, by = "sampleName")
  }

  return(data)
}
```
The same can be applied to other two readers.